kmyth_log/src/kmyth_log.c

```diff
@@ -0,0 +1,350 @@
+
+  // if application name was truncated, notify user 
+  if (truncated == true)
+  {
+    fprintf(stderr, "set_app_version(): input \"%s\" ", new_app_version);
+    fprintf(stderr, "truncated to \"%s\"\n", log_settings.app_version);
+  }
+}
+
+//############################################################################
+// set_applog_path()
+//############################################################################
+void set_applog_path(char *new_applog_path)
+{
+  if (strlen(new_applog_path) <= MAX_APPLOG_PATH_LEN)
+  {
+    log_settings.applog_path_len = strlen(new_applog_path);
+    strncpy(log_settings.applog_path,
+            new_applog_path, log_settings.applog_path_len);
+
+    // ensure log directory string is null terminated
+    log_settings.applog_path[log_settings.applog_path_len] = '\0';
+  }
+  else
+  {
+    // do nothing if path string is too long, but warn user
+    fprintf(stderr, "set_applog_path(): ");
+    fprintf(stderr, "input \"%s\" exceeds maximum length ", new_applog_path);
+    fprintf(stderr, "(%d) - application log directory ", MAX_APPLOG_PATH_LEN);
+    fprintf(stderr, "remains \"%s\"\n", log_settings.applog_path);
+  }
+}
+
+//############################################################################
+// set_applog_output_mode()
+//   - valid values: 0, 1, or 2
+//############################################################################
+void set_applog_output_mode(int new_output_mode)
+{
+  if ((new_output_mode >= 0) && (new_output_mode <= 2))
+  {
+    log_settings.applog_output_mode = new_output_mode;
+  }
+  else
+  {
+    // do nothing if invalid, but warn user
+    fprintf(stderr, "set_applog_output_mode(): ");
+    fprintf(stderr, "input (%d) invalid ", new_output_mode);
+    fprintf(stderr, "- unchanged (%d)\n", log_settings.applog_output_mode);
+  }
+}
+
+//############################################################################
+// set_applog_severity_threshold()
+//   - valid values: 0-7
+//############################################################################
+void set_applog_severity_threshold(int new_severity_threshold)
+{
+  if ((new_severity_threshold >= 0) && (new_severity_threshold <= 7))
+  {
+    log_settings.applog_severity_threshold = new_severity_threshold;
+  }
+  else
+  {
+    // do nothing if invalid, but warn user
+    fprintf(stderr, "set_applog_severity_threshold(): ");
+    fprintf(stderr, "input (%d) invalid - unchanged ", new_severity_threshold);
+    fprintf(stderr, "(%d)\n", log_settings.applog_severity_threshold);
+  }
+}
+
+//############################################################################
+// set_syslog_facility()
+//   - valid values: (0<<3) - ((LOG_NFACILITIES-1)<<3)
+//############################################################################
+void set_syslog_facility(int new_syslog_facility)
+{
+  if ((LOG_FAC(new_syslog_facility) >= 0) &&
+      (LOG_FAC(new_syslog_facility) <= (LOG_NFACILITIES - 1)))
```

Comment:
This doesn't appear to be consistent with the comment, which suggests the (LOG_NFACILITIES-1) should be left shifted 3.

Reply:
Because LOG_FAC() right shifts new_syslog_facility by three bits, I think this is consistent. If the comments do not make this clearer or if you still believe this is wrong, though, let me know and I will look at it again.